docs/testplan/server-side-copy-testplan.md

```diff
@@ -0,0 +1,53 @@
+|                        | ~~Impact on Metainfo > Expired Deletion                         | While metainfo loop is going through expired deletion, it should be possible to server side copy with no negative impact on former services~~                                                                                                                             | This won't be possible as currently designed, as storage nodes will delete the pieces on the original expiration date.         |
+|                        | ~~Impact on Accounting/Tally                                    | While accounting services are running, it should be possible to server side copy with no negative impact on former services~~                                                                                                                                             | This won't be possible as currently designed, as storage nodes will delete the pieces on the original expiration date.         |
+|                        | ~~Expiration Dates                                              | For server side copy users are allowed to make new copies of pieces which allows users to change their expiration dates, so in this way users are able to change expiration date of the new object through server side copy~~                                             | This won't be possible as currently designed, as storage nodes will delete the pieces on the original expiration date.         |
+| DB- Table Segment      | Expiration Date                                               | If a user uses Server-side copy, then the source object and the destination object must have the same experation date                                                                                                                                                   |          |
+|                        | ancestor_stream_id negative                                   | If a segment hasn't been copied, then it is empty under ancestor_stream_id in the db                                                                                                                                                                                    |          |
+|                        | ancestor_stream_id positive                                   | If a segment has been copied, then for the copied segment it contains the ancestor_stream_id which is the stream_id for the original segment                                                                                                                            |          |
+|                        | reference_counter integer                                     | there is a reference_counter integer which counts the number of segments having stream_id as ancestor_stream_id, so If a new segment has been server side copied then the integer should reflect that (+1)                                                              |          |
```

Comment:
We should add the edge case where we already have max(reference_counter) copies. We should throw an error in such case.

Reply:
Updated